src/parser/jobs/blm/Leylines.js

```diff
@@ -68,26 +141,72 @@ export default class Leylines extends Module {
 			}
 			this._circleOfPowers.history.push(this._circleOfPowers.current)
 		}
+		if (this._leyLines.current) {
+			if (!this._leyLines.current.stop) {
+				this._leyLines.current.stop = event.timestamps
+			}
+			this._leyLines.history.push(this._leyLines.current)
+		}
 		//check if there even were any events
 		if (!this._circleOfPowers.history.length) { return }
-		//filter out the last possible LL usage because it would make things weird.
-		this._circleOfPowers.history = this._circleOfPowers.history.filter(cops => cops.start < (event.timestamp - LEYLINE_DURATION))
-		this._leyLineHistory = this._leyLineHistory.filter(timestamps => timestamps < (event.timestamp - LEYLINE_DURATION))
+		if (!this._leyLines.history.length) { return }
+
+		// Get the total duration of CoP uptime and ley lines, so we can get the overall percentage uptime
 		const circleOfPowerDurations = this._circleOfPowers.history.map(cops => Math.max(cops.stop - cops.start, 0))
 		const sumOfCoPUpTime = circleOfPowerDurations.reduce((accumulator, currentValue) => accumulator + currentValue, 0)
-		const numberOfLeyLines = this._leyLineHistory.length
+		const leyLinesDurations = this._leyLines.history.map(lines => Math.max(lines.stop - lines.start, 0))
+		const sumOfLeyLineDurations = leyLinesDurations.reduce((accumulator, currentValue) => accumulator + currentValue, 0)
 
 		this.checklist.add(new Rule({
 			name: <Trans id="blm.leylines.checklist-caption">Stay in your Ley Lines</Trans>,
 			description: <Trans id="blm.leylines.checklist">Try to avoid leaving your Ley Lines after placing them. Take advantage of Ley Lines' size to stay in them while dodging AOEs and being in range of healers. If you can't stay in them for the majority of a Ley Lines' duration, consider changing where they're placed in the fight.</Trans>,
 			requirements: [
 				new Requirement({
 					name: <ActionLink {...ACTIONS.LEY_LINES} />,
-					percent: this._percentFunction(numberOfLeyLines, sumOfCoPUpTime),
+					percent: this._percentFunction(sumOfLeyLineDurations, sumOfCoPUpTime),
 				}),
 			],
 			//pretty random. Should be revised, maybe based on fights? 10% is ~ 1 GCD. So we allow that.
 			target: 90,
 		}))
 	}
+
+	output() {
+		const panels = this._leyLines.history.map(leyLinesEvent => {
```

Comment:
Given the content is just the timeline link now, accordion might be overkill for the job. Perhaps a list of links or similar? Not sure.